File: classic puzzle-medium/ip-mask-calculating.py
#https://www.codingame.com/training/medium/ip-mask-calculating
#IP MASK CALCULATING
ip=input().split("/")
s=int(ip[1])*'1'+(32-int(ip[1]))*'0'
mask=[int(s[x:x+8],2) for x in range(0,len(s),8)]
net=[int(x) for x in ip[0].split('.')]
subnet=[str(int(bin(ioctet & moctet), 2)) for ioctet, moctet in zip(net, mask)]
broadcast=[str((ioctet | ~moctet) & 0xff) for ioctet, moctet in zip(net, mask)]
print('.'.join(subnet))
print('.'.join(broadcast))

File: classic puzzle-medium/ip-mask-calculating.js
//https://www.codingame.com/training/medium/ip-mask-calculating
//IP MASK CALCULATING
const ip = readline().split("/");
var s='1'.repeat(ip[1])+'0'.repeat(32-ip[1]);
var mask=[];
for(let i=0;i<s.length;i+=8)
    mask.push(parseInt(s.slice(i,i+8),2));
var boardcast=[];
var net=ip[0].split('.').map(Number).map((e, i)=>{
    boardcast.push(((e | ~mask[i]) & 0xff));
    return (e & mask[i]);
});
console.log(net.map(String).join('.'));
console.log(boardcast.map(String).join('.'));
